proposals/0183-skip-rent-rewrites.md

```diff
@@ -0,0 +1,69 @@
+---
+simd: '0183'
+title: Skip Rent Rewrites
+authors:
+  - brooks@anza.xyz
+category: Standard
+type: Core
+status: Draft
+created: 2024-10-04
+feature: https://github.com/solana-labs/solana/issues/26599
+---
+
+## Summary
+
+Do not rewrite accounts *that are unchanged* by rent collection.
+
+## Motivation
+
+Rent collection checks every account at least once per epoch.  This process
+loads *and stores* accounts, regardless if the account has actually changed or
+not.  All accounts now must be rent-exempt, and there are zero rent-paying
+accounts on mainnet-beta.  Thus, almost every account stored due to rent
+collection is unchanged, and storing unchanged accounts is useless work.
+
+Worse, storing accounts incurs all the downstream costs on the accounts
+database: computing the various accounts hashes on extra accounts unnecessarily,
+tracking/cleaning/purging the new and old versions of accounts, plus it bloats
+the incremental snapshots with accounts that haven't meaningfully changed.
+
+## Alternatives Considered
+
+We could remove rent collection entirely, and that is already planned
+(see SIMD-0084).  Skipping rewrites is a smaller, less complex change, which
+can allow rollout and activation sooner.
+
+## New Terminology
+
+"Skipping rewrites" means to not store accounts that are unchanged by rent
+collection.
+
+## Detailed Design
+
+If rent collection has *not* changed an account, the account must not be
+stored/written back in that slot.
+
+An important note, rent collection updates an account's `rent epoch` in addition
+to its balance.  If rent collection does not collect rent (i.e. the account's
+balance is unchanged), but *does* change the account's `rent epoch`, the
+account must still be stored.
+
+## Impact
+
```

Comment:
maybe, mention that account's delta hash and incremental snapshot hash will change, but full hash won't change?